src/map/lua/lua_baseentity.cpp

```diff
@@ -711,6 +711,30 @@ void CLuaBaseEntity::setLocalVar(std::string const& var, uint32 val)
     m_PBaseEntity->SetLocalVar(var.c_str(), val);
 }
 
+/************************************************************************
+ *  Function: clearLocalVarsWithPrefix()
+ *  Purpose : Deletes all local variables with the given prefix.
+ *  Example : pet:clearLocalVarsWithPrefix("volt");
+ *  Notes   :
+ ************************************************************************/
+
+void CLuaBaseEntity::clearLocalVarsWithPrefix(std::string const& prefix)
+{
+    auto& localVars = m_PBaseEntity->GetLocalVars();
+
+    auto iter = localVars.begin();
+    while (iter != localVars.end())
+    {
+        if (iter->first.rfind(prefix, 0) == 0)
+        {
+            m_PBaseEntity->SetLocalVar(iter->first.c_str(), 0);
+        }
+        ++iter;
+    }
```

Comment:
```cpp
for (const auto& localVar : m_PBaseEntity->GetLocalVars())
{
    if (starts_with(localVar, prefix))
    {
        m_PBaseEntity->SetLocalVar(localVar, 0);
    }
}
```

Reply:
Ah didnt see it was pairs, use: https://en.cppreference.com/w/cpp/language/structured_binding

Reply:
```cpp
for (const auto& [localVar, _] : m_PBaseEntity->GetLocalVars())
{
    if (starts_with(localVar, prefix))
    {
        m_PBaseEntity->SetLocalVar(localVar, 0);
    }
}
```